core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -707,7 +707,8 @@ public MergeWriterNode plan(Merge merge)
                         String columnName = columnSchema.getName();
                         Expression coalesced = new CoalesceExpression(rewritten, new Cast(failFunction(metadata, session, INVALID_ARGUMENTS, "Assigning NULL to non-null MERGE target table column " + columnName), toSqlType(columnSchema.getType())));
                         rowBuilder.add(coalesced);
-                    } else {
+                    }
+                    else {
```

Comment:
belongs to the first commit

Reply:
strange that checkstyle didn't complain :/

Reply:
anyway there will be no else

Reply:
> strange that checkstyle didn't complain :/

that's something we can try to improve. do you want to take a stab?